packages/postgresql/src/Adaptor.js

```diff
@@ -307,30 +325,37 @@ export function insert(table, record, options, callback) {
 export function insertMany(table, records, options, callback) {
   return state => {
     let { client } = state;
-
+    const [resolvedTable, resolvedRecords, resolvedOptions] = expandReferences(
+      state,
+      table,
+      records,
+      options
+    );
     try {
-      const data = expandReferences(records)(state);
-
       return new Promise((resolve, reject) => {
-        if (!data || data.length === 0) {
+        if (!resolvedRecords || resolvedRecords.length === 0) {
           console.log('No records provided; skipping insert.');
           resolve(state);
         }
         // Note: we select the keys of the FIRST object as the canonical template.
-        const columns = Object.keys(data[0]);
+        const columns = Object.keys(resolvedRecords[0]);
         const columnsList = columns.join(', ');
-        const valueSets = data.map(x => Object.values(x));
+        const valueSets = resolvedRecords.map(x => Object.values(x));
 
         const query = handleValues(
-          format(`INSERT INTO ${table} (${columnsList}) VALUES %L;`, valueSets),
-          handleOptions(options)
+          format(
+            `INSERT INTO ${resolvedTable} (${columnsList}) VALUES %L;`,
+            valueSets
+          ),
+          handleOptions(resolvedOptions)
         );
 
-        const safeQuery = `INSERT INTO ${table} (${columnsList}) VALUES [--REDACTED--]];`;
+        const safeQuery = `INSERT INTO ${resolvedTable} (${columnsList}) VALUES [--REDACTED--]];`;
 
-        const queryToLog = options && options.logValues ? query : safeQuery;
+        const queryToLog =
+          resolvedOptions && resolvedOptions.logValues ? query : safeQuery;
```

Comment:
Again, as we're touching it anyway, `resolvedOptions.logValues ? query : safeQuery;`